File: Data Structures & Algorithms/non-overlapping-intervals/submission-1.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(),intervals.end());

        // vector<vector<int>> ans;
        int ans = 0;
        int prevEnd =  intervals[0][1];
        vector<int> interval = intervals[0];
        for(int i=1;i<intervals.size();i++){
        
            if(prevEnd<=intervals[i][0]){
                // ans.push_back(interval);
                prevEnd = intervals[i][1];
            }
            else{
                // interval[1] = max(intervals[i][1],interval[1]);
                ans++;
                prevEnd = min(prevEnd,intervals[i][1]);
            }
        }

        // ans.push_back(interval);
        return ans;
    }
};

File: Data Structures & Algorithms/jump-game-ii/submission-1.py
class Solution:
    def jump(self, nums: List[int]) -> int:

        rangee = 0
        dr = 0
        ans = 0
        for i in range(len(nums)):
            n = nums[i]
            rangee = max(rangee,n)
            if dr == 0 and i<(len(nums)-1):
                dr = rangee
                ans +=1
            rangee-=1
            dr -= 1
        return ans